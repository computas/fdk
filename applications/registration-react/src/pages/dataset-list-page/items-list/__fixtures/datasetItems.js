export const datasetItems = {
  isFetchingDatasets: false,
  datasetItems: [
    {
      id: '084f747f-04a9-460e-b07e-ee74c60a2086',
      uri:
        'http://brreg.no/catalogs/910244132/datasets/084f747f-04a9-460e-b07e-ee74c60a2086',
      title: {
        nb: 'test ois'
      },
      description: {
        nb: 'tester'
      },
      objective: {
        nb: ''
      },
      publisher: {
        uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
        name: 'RAMSUND OG ROGNAN REVISJON'
      },
      type: 'Data',
      landingPage: [''],
      theme: [
        {
          uri: '',
          title: {
            nb: ''
          }
        },
        {
          uri:
            'http://publications.europa.eu/resource/authority/data-theme/AGRI',
          title: {
            nb: 'Jordbruk, fiskeri, skogbruk og mat'
          }
        }
      ],
      accessRights: {
        uri:
          'http://publications.europa.eu/resource/authority/access-right/NON_PUBLIC',
        prefLabel: {}
      },
      legalBasisForRestriction: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'skjerming'
          },
          extraType: null
        }
      ],
      legalBasisForProcessing: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'behandling'
          },
          extraType: null
        }
      ],
      legalBasisForAccess: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'utlevering'
          },
          extraType: null
        }
      ],
      catalogId: '910244132',
      _lastModified: '2018-05-07T10:36:20.388+0000',
      registrationStatus: 'PUBLISH',
      subject: [
        {
          uri: 'https://data-david.github.io/Begrep/begrep/Enhet',
          identifier: 'https://data-david.github.io/Begrep/begrep/Enhet',
          prefLabel: {
            no: 'enhet'
          },
          altLabel: null,
          definition: {
            no: 'alt som er registrert med et organisasjonsnummer '
          },
          note: {
            no:
              'Alle hovedenheter, underenheter og organisasjonsledd som er identifisert med et organisasjonsnummer.'
          },
          source: 'https://jira.brreg.no/browse/BEGREP-208',
          creator: {
            uri: 'http://data.brreg.no/enhetsregisteret/enhet/974760673'
          },
          inScheme: ['http://data-david.github.io/vokabular/Befolkning'],
          datasets: null
        }
      ],
      keyword: [
        {
          nb: 'søkeord'
        }
      ],
      contactPoint: [
        {
          email: 'epost@kontakskjema.no',
          hasTelephone: '12345678',
          hasURL: 'https://www.google.no',
          organizationUnit: 'Kontaktpunkt'
        }
      ],
      conformsTo: [
        {
          uri: '',
          prefLabel: {
            nb: 'test'
          },
          extraType: null
        }
      ],
      references: [
        {
          referenceType: {
            uri: 'dct:isPartOf',
            code: 'isPartOf',
            prefLabel: {}
          },
          source: {
            uri:
              'http://brreg.no/catalogs/910244132/datasets/0aa1861a-9af0-4388-a81e-6b05de0fb63e',
            prefLabel: null,
            extraType: null
          }
        }
      ],
      hasRelevanceAnnotation: {
        inDimension: 'iso:Relevance',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Relevans'
        }
      },
      hasCompletenessAnnotation: {
        inDimension: 'iso:Completeness',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Kompletthet'
        }
      },
      hasAccuracyAnnotation: {
        inDimension: 'iso:Accuracy',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Nøyaktighet'
        }
      },
      hasAvailabilityAnnotation: {
        inDimension: 'iso:Availability',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Tilgjengelighet'
        }
      },
      distribution: [
        {
          id: '',
          description: {
            nb: 'Beskrivelse av distribusjon'
          },
          accessURL: ['http://google.no'],
          license: {
            uri: 'http://creativecommons.org/publicdomain/zero/1.0/',
            prefLabel: null,
            extraType: null
          },
          conformsTo: [
            {
              prefLabel: {
                nb: 'Standard'
              },
              uri: 'http://google.no'
            }
          ],
          page: [
            {
              uri: 'https://google.com',
              prefLabel: {
                nb: ''
              },
              extraType: null
            }
          ],
          format: ['json', 'xml'],
          type: 'API'
        }
      ],
      sample: [
        {
          id: '',
          description: {
            nb: ''
          },
          accessURL: [],
          license: {
            uri: '',
            prefLabel: {
              nb: ''
            },
            extraType: null
          },
          conformsTo: [],
          page: [
            {
              uri: '',
              prefLabel: {
                nb: ''
              },
              extraType: null
            }
          ],
          format: [],
          type: 'API'
        }
      ],
      openLicenseItems: [
        {
          uri: 'http://creativecommons.org/licenses/by/4.0/',
          code: 'CC BY 4.0',
          prefLabel_no: 'Creative Commons Navngivelse 4.0 Internasjonal'
        },
        {
          uri: 'http://creativecommons.org/publicdomain/zero/1.0/',
          code: 'CC0 1.0',
          prefLabel_no: 'Creative Commons Universal Fristatus-erklæring'
        },
        {
          uri: 'http://data.norge.no/nlod/',
          code: 'NLOD',
          prefLabel_no: 'Norsk lisens for offentlige data'
        }
      ],
      informationModel: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'test'
          },
          extraType: null
        }
      ],
      frequencyItems: null,
      provenanceItems: null,
      provenance: {},
      modified: null,
      hasCurrentnessAnnotation: {
        inDimension: 'iso:Currentness',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Aktualitet'
        }
      },
      accrualPeriodicity: {
        uri:
          'http://publications.europa.eu/resource/authority/frequency/ANNUAL',
        code: 'ANNUAL',
        prefLabel: {}
      },
      spatial: [
        {
          uri: 'Oslo'
        }
      ],
      temporal: [{}],
      issued: null,
      language: []
    }
  ]
};
