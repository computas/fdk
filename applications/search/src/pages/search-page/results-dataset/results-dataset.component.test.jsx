import React from 'react';
import { shallow } from 'enzyme';
import { ResultsDataset } from './results-dataset.component';
import datasetItems from '../../../../test/fixtures/datasetItems';

let closeFilterModal;
let onFilterTheme;
let onFilterAccessRights;
let onFilterPublisherHierarchy;
let onClearSearch;
let onSort;
let onPageChange;
let defaultProps;
let wrapper;

beforeEach(() => {
  closeFilterModal = jest.fn();
  onFilterTheme = jest.fn();
  onFilterAccessRights = jest.fn();
  onFilterPublisherHierarchy = jest.fn();
  onClearSearch = jest.fn();
  onSort = jest.fn();
  onPageChange = jest.fn();

  defaultProps = {
    onSort
  };
  defaultProps = {
    showFilterModal: true,
    datasetItems: {},
    searchQuery: {},
    themesItems: {},
    publisherArray: [],
    publishers: {},
    showClearFilterButton: false,
    hitsPerPage: 10,
    closeFilterModal,
    onFilterTheme,
    onFilterAccessRights,
    onFilterPublisherHierarchy,
    onClearSearch,
    onSort,
    onPageChange
  };
  wrapper = shallow(<ResultsDataset {...defaultProps} />);
});

test('should render ResultsDataset correctly with minimum of props', () => {
  const minWrapper = shallow(<ResultsDataset onSort={onSort} />);
  expect(minWrapper).toMatchSnapshot();
});

test('should render ResultsDataset correctly with props', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ResultsDataset correctly with hits', () => {
  wrapper.setProps({
    datasetItems: datasetItems[0]
  });
  expect(wrapper).toMatchSnapshot();
});
