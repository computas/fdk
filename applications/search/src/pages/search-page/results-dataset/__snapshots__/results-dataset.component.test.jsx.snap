// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultsDataset correctly with hits 1`] = `
<main
  id="content"
>
  <section
    className="row mb-3"
  >
    <div
      className="col-6 col-lg-4"
    >
      <button
        className="btn btn-primary fdk-button fade-in-500 d-none"
        onClick={[MockFunction]}
        type="button"
      >
        Clear
      </button>
    </div>
    <div
      className="col-6 col-lg-4 offset-lg-4"
    >
      <div
        className="float-right"
      >
        <Select
          activeSort={null}
          items={
            Array [
              Object {
                "defaultOption": true,
                "field": "_score",
                "label": "relevance",
                "order": "asc",
              },
              Object {
                "field": "title",
                "label": "title",
                "order": "asc",
              },
              Object {
                "field": "modified",
                "label": "modified",
                "order": "desc",
              },
              Object {
                "field": "publisher.name",
                "label": "publisher",
                "order": "asc",
              },
            ]
          }
          onChange={[MockFunction]}
        />
      </div>
    </div>
  </section>
  <section
    className="row"
  >
    <aside
      className="search-filters col-lg-4 d-none d-lg-block"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Filtrering tilgang
      </span>
      <div>
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={false}
          fade={true}
          isOpen={true}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          role="dialog"
          toggle={[MockFunction]}
          zIndex={1050}
        >
          <ModalHeader
            closeAriaLabel="Close"
            tag="h5"
            toggle={[MockFunction]}
            wrapTag="div"
          >
            Filter
          </ModalHeader>
          <ModalBody
            tag="div"
          >
            <div
              className="search-filters"
            >
              <FilterBox
                activeFilter={null}
                filter={
                  Object {
                    "buckets": Array [
                      Object {
                        "count": 1,
                        "key": "AGRI",
                      },
                      Object {
                        "count": 1,
                        "key": "ECON",
                      },
                      Object {
                        "count": 1,
                        "key": "EDUC",
                      },
                      Object {
                        "count": 1,
                        "key": "ENER",
                      },
                      Object {
                        "count": 1,
                        "key": "ENVI",
                      },
                      Object {
                        "count": 1,
                        "key": "GOVE",
                      },
                      Object {
                        "count": 1,
                        "key": "HEAL",
                      },
                      Object {
                        "count": 1,
                        "key": "INTR",
                      },
                      Object {
                        "count": 1,
                        "key": "JUST",
                      },
                      Object {
                        "count": 1,
                        "key": "REGI",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  }
                }
                htmlKey={1}
                onClick={[MockFunction]}
                themesItems={Object {}}
                title="Theme"
              />
              <FilterBox
                activeFilter={null}
                filter={
                  Object {
                    "buckets": Array [
                      Object {
                        "count": 1,
                        "key": "PUBLIC",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  }
                }
                htmlKey={2}
                onClick={[MockFunction]}
                themesItems={null}
                title="Access"
              />
              <SearchPublishersTree
                activeFilter={null}
                filter={Array []}
                onFilterPublisherHierarchy={[MockFunction]}
                onSearch={null}
                orgPath={null}
                publishers={Object {}}
                title="Organization"
              />
              <FilterBox
                activeFilter={null}
                filter={
                  Object {
                    "buckets": Array [
                      Object {
                        "count": 1,
                        "key": "Ukjent",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  }
                }
                htmlKey={3}
                onClick={null}
                themesItems={null}
                title="Spatial"
              />
              <FilterBox
                activeFilter={null}
                filter={
                  Object {
                    "buckets": Array [
                      Object {
                        "count": 1,
                        "key": "VEDTAK",
                      },
                    ],
                    "doc_count_error_upper_bound": 0,
                    "sum_other_doc_count": 0,
                  }
                }
                htmlKey={4}
                onClick={null}
                themesItems={null}
                title="Provenance"
              />
            </div>
          </ModalBody>
          <ModalFooter
            tag="div"
          >
            <Button
              className="fdk-button"
              color="primary"
              onClick={[MockFunction]}
              tag="button"
            >
              Close
            </Button>
          </ModalFooter>
        </Modal>
        <FilterBox
          activeFilter={null}
          filter={
            Object {
              "buckets": Array [
                Object {
                  "count": 1,
                  "key": "AGRI",
                },
                Object {
                  "count": 1,
                  "key": "ECON",
                },
                Object {
                  "count": 1,
                  "key": "EDUC",
                },
                Object {
                  "count": 1,
                  "key": "ENER",
                },
                Object {
                  "count": 1,
                  "key": "ENVI",
                },
                Object {
                  "count": 1,
                  "key": "GOVE",
                },
                Object {
                  "count": 1,
                  "key": "HEAL",
                },
                Object {
                  "count": 1,
                  "key": "INTR",
                },
                Object {
                  "count": 1,
                  "key": "JUST",
                },
                Object {
                  "count": 1,
                  "key": "REGI",
                },
              ],
              "doc_count_error_upper_bound": 0,
              "sum_other_doc_count": 0,
            }
          }
          htmlKey={1}
          onClick={[MockFunction]}
          themesItems={Object {}}
          title="Theme"
        />
        <FilterBox
          activeFilter={null}
          filter={
            Object {
              "buckets": Array [
                Object {
                  "count": 1,
                  "key": "PUBLIC",
                },
              ],
              "doc_count_error_upper_bound": 0,
              "sum_other_doc_count": 0,
            }
          }
          htmlKey={2}
          onClick={[MockFunction]}
          themesItems={null}
          title="Access"
        />
        <SearchPublishersTree
          activeFilter={null}
          filter={Array []}
          onFilterPublisherHierarchy={[MockFunction]}
          onSearch={null}
          orgPath={null}
          publishers={Object {}}
          title="Organization"
        />
        <FilterBox
          activeFilter={null}
          filter={
            Object {
              "buckets": Array [
                Object {
                  "count": 1,
                  "key": "Ukjent",
                },
              ],
              "doc_count_error_upper_bound": 0,
              "sum_other_doc_count": 0,
            }
          }
          htmlKey={3}
          onClick={null}
          themesItems={null}
          title="Spatial"
        />
        <FilterBox
          activeFilter={null}
          filter={
            Object {
              "buckets": Array [
                Object {
                  "count": 1,
                  "key": "VEDTAK",
                },
              ],
              "doc_count_error_upper_bound": 0,
              "sum_other_doc_count": 0,
            }
          }
          htmlKey={4}
          onClick={null}
          themesItems={null}
          title="Provenance"
        />
      </div>
    </aside>
    <section
      className="col-12 col-lg-8"
    >
      <ErrorBoundary
        key="b12e9e8d-3833-49e2-9f62-0f8fd877be05"
      >
        <SearchHitItem
          distributionTypeItems={null}
          result={
            Object {
              "_id": "b12e9e8d-3833-49e2-9f62-0f8fd877be05",
              "_index": "dcat",
              "_score": null,
              "_source": Object {
                "accessRights": Object {
                  "code": "PUBLIC",
                  "prefLabel": Object {
                    "en": "Public",
                    "nb": "Offentlig",
                    "nn": "Offentlig",
                  },
                  "uri": "http://publications.europa.eu/resource/authority/access-right/PUBLIC",
                },
                "catalog": Object {
                  "id": "910244132",
                  "publisher": Object {
                    "id": "910244132",
                    "name": "RAMSUND OG ROGNAN REVISJON",
                    "orgPath": "/ANNET/910244132",
                    "type": "no.dcat.datastore.domain.dcat.Publisher",
                    "uri": "http://data.brreg.no/enhetsregisteret/enhet/910244132",
                    "valid": false,
                  },
                  "title": Object {
                    "nb": "Datakatalog for RAMSUND OG ROGNAN REVISJON",
                  },
                  "uri": "http://brreg.no/catalogs/910244132",
                },
                "contactPoint": Array [],
                "description": Object {
                  "nb": "test",
                },
                "descriptionFormatted": Object {
                  "nb": "test",
                },
                "harvest": Object {
                  "changed": Array [
                    "2018-04-13T08:26:27+0200",
                  ],
                  "firstHarvested": "2018-04-12T17:18:30+0200",
                  "lastChanged": "2018-04-13T08:26:27+0200",
                  "lastHarvested": "2018-04-13T08:26:27+0200",
                },
                "id": "b12e9e8d-3833-49e2-9f62-0f8fd877be05",
                "landingPage": Array [],
                "provenance": Object {
                  "code": "NASJONAL",
                  "prefLabel": Object {
                    "en": "National Building Block",
                    "nb": "Nasjonal felleskomponent",
                    "nn": "Nasjonal felleskomponent",
                  },
                  "uri": "http://data.brreg.no/datakatalog/provinens/nasjonal",
                },
                "publisher": Object {
                  "id": "910244132",
                  "name": "RAMSUND OG ROGNAN REVISJON",
                  "orgPath": "/ANNET/910244132",
                  "type": "no.dcat.datastore.domain.dcat.Publisher",
                  "uri": "http://data.brreg.no/enhetsregisteret/enhet/910244132",
                  "valid": false,
                },
                "source": "A",
                "theme": Array [
                  Object {
                    "code": "EDUC",
                    "conceptSchema": Object {
                      "id": "http://publications.europa.eu/resource/authority/data-theme",
                      "title": Object {
                        "en": "Dataset types Named Authority List",
                      },
                      "versioninfo": "20160921-0",
                      "versionnumber": "20160921-0",
                    },
                    "id": "http://publications.europa.eu/resource/authority/data-theme/EDUC",
                    "startUse": "2015-10-01",
                    "title": Object {
                      "bg": "Образование, култура и спорт",
                      "cs": "Vzdělávání, kultura a sport",
                      "da": "Uddannelse, kultur og sport",
                      "de": "Bildung, Kultur und Sport",
                      "el": "Παιδεία, πολιτιστικά θέματα και αθλητισμός",
                      "en": "Education, culture and sport",
                      "es": "Educación, cultura y deportes",
                      "et": "Haridus, kultuur ja sport",
                      "fi": "Koulutus, kulttuuri ja urheilu",
                      "fr": "Éducation, culture et sport",
                      "ga": "Oideachas, cultúr agus spórt",
                      "hr": "Obrazovanje, kultura i sport",
                      "hu": "Oktatás, kultúra és sport",
                      "it": "Istruzione, cultura e sport",
                      "lt": "Švietimas, kultūra ir sportas",
                      "lv": "Izglītība, kultūra un sports",
                      "mt": "Edukazzjoni, kultura u sport",
                      "nb": "Utdanning, kultur og sport",
                      "nl": "Onderwijs, cultuur en sport",
                      "pl": "Edukacja, kultura i sport",
                      "pt": "Educação, cultura e desporto",
                      "ro": "Educaţie, cultură şi sport",
                      "sk": "Vzdelávanie, kultúra a šport",
                      "sl": "Izobraževanje, kultura in šport",
                      "sv": "Utbildning, kultur och sport",
                    },
                  },
                ],
                "title": Object {
                  "nb": "test ois",
                },
                "uri": "http://brreg.no/catalogs/910244132/datasets/633a0af6-5c88-4027-9f58-79f487a80642",
              },
              "_type": "dataset",
              "sort": Array [
                "nasjonal",
                "a",
                1523600787000,
              ],
            }
          }
        />
      </ErrorBoundary>
    </section>
    <section
      className="col-12 col-lg-8 offset-lg-4 d-flex justify-content-center"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Sidepaginering.
      </span>
      <PaginationBoxView
        activeClassName="active"
        breakClassName="break-me"
        breakLabel={
          <span>
            ...
          </span>
        }
        containerClassName="pagination"
        disableInitialCallback={true}
        disabledClassName="disabled"
        initialPage={0}
        marginPagesDisplayed={1}
        nextClassName="next"
        nextLabel="Neste side"
        onPageChange={[MockFunction]}
        pageCount={14}
        pageRangeDisplayed={2}
        previousClassName="previous"
        previousLabel="Forrige side"
        subContainerClassName="pages pagination"
      />
    </section>
  </section>
</main>
`;

exports[`should render ResultsDataset correctly with minimum of props 1`] = `
<main
  id="content"
>
  <section
    className="row mb-3"
  >
    <div
      className="col-6 col-lg-4"
    >
      <button
        className="btn btn-primary fdk-button fade-in-500 d-none"
        onClick={null}
        type="button"
      >
        Clear
      </button>
    </div>
    <div
      className="col-6 col-lg-4 offset-lg-4"
    >
      <div
        className="float-right"
      >
        <Select
          activeSort={null}
          items={
            Array [
              Object {
                "defaultOption": true,
                "field": "_score",
                "label": "relevance",
                "order": "asc",
              },
              Object {
                "field": "title",
                "label": "title",
                "order": "asc",
              },
              Object {
                "field": "modified",
                "label": "modified",
                "order": "desc",
              },
              Object {
                "field": "publisher.name",
                "label": "publisher",
                "order": "asc",
              },
            ]
          }
          onChange={[MockFunction]}
        />
      </div>
    </div>
  </section>
  <section
    className="row"
  >
    <aside
      className="search-filters col-lg-4 d-none d-lg-block"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Filtrering tilgang
      </span>
    </aside>
    <section
      className="col-12 col-lg-8"
    />
  </section>
</main>
`;

exports[`should render ResultsDataset correctly with props 1`] = `
<main
  id="content"
>
  <section
    className="row mb-3"
  >
    <div
      className="col-6 col-lg-4"
    >
      <button
        className="btn btn-primary fdk-button fade-in-500 d-none"
        onClick={[MockFunction]}
        type="button"
      >
        Clear
      </button>
    </div>
    <div
      className="col-6 col-lg-4 offset-lg-4"
    >
      <div
        className="float-right"
      >
        <Select
          activeSort={null}
          items={
            Array [
              Object {
                "defaultOption": true,
                "field": "_score",
                "label": "relevance",
                "order": "asc",
              },
              Object {
                "field": "title",
                "label": "title",
                "order": "asc",
              },
              Object {
                "field": "modified",
                "label": "modified",
                "order": "desc",
              },
              Object {
                "field": "publisher.name",
                "label": "publisher",
                "order": "asc",
              },
            ]
          }
          onChange={[MockFunction]}
        />
      </div>
    </div>
  </section>
  <section
    className="row"
  >
    <aside
      className="search-filters col-lg-4 d-none d-lg-block"
    >
      <span
        aria-hidden="false"
        className="uu-invisible"
      >
        Filtrering tilgang
      </span>
    </aside>
    <section
      className="col-12 col-lg-8"
    />
  </section>
</main>
`;
